app/main.py

```diff
@@ -1,8 +1,27 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+        Person.people[name] = self
+
+
+def create_person_list(people_list: list) -> list:
+    new_people_list = [Person(person["name"], person["age"])
+                       for person in people_list]
+
+    for person in people_list:
+        if "husband" in person and person["husband"]:
+            if person["name"] in Person.people:
+                husband = Person.people[person["husband"]]
+                wife = Person.people[person["name"]]
+                wife.husband = husband
+
+        elif "wife" in person and person["wife"]:
+            if person["name"] in Person.people:
+                wife = Person.people[person["wife"]]
+                husband = Person.people[person["name"]]
+                husband.wife = wife
```

Comment:
You could simplify this condition in the following way:
```suggestion
    for person in people_list:
        person_instance = Person.people[person["name"]]
        if husband_name := person.get("husband"):
            person_instance.husband = Person.people[husband_name]
        elif wife_name := person.get("wife"):
             person_instance.wife = Person.people[wife_name]
```
Modify this approach in case there are errors during the program execution but keep the overall structure.